Client/Frontend/Browser/PageZoom/PageZoomSettingsView.swift

```diff
@@ -0,0 +1,170 @@
+        Section(
+          header: Text(Strings.PageZoom.specificWebsitesZoomLevelsSectionTitle)
+            .font(.footnote)
+            .foregroundColor(Color(UIColor.secondaryBraveLabel))
+        ) {
+          ForEach(domains) { domain in
+            HStack {
+              let url = domain.url ?? ""
+              
+              let zoomLevel = domain.zoom_level?.doubleValue ?? 1.0
+              
+              let percentageLevel = Int(zoomLevel * 100.0)
+              
+              Text(url)
+                .font(.headline)
+                .foregroundColor(Color(UIColor.braveLabel))
+              
+              Spacer()
+              
+              Text(PageZoomView.percentFormatter.string(from: .init(value: zoomLevel)) ?? "\(percentageLevel)%")
+                .foregroundColor(Color(UIColor.braveLabel))
+                .multilineTextAlignment(.trailing)
+            }
+          }
+          .onDelete(perform: resetItemZoomLevel)
+        }
+        .listRowBackground(Color(UIColor.secondaryBraveGroupedBackground))
+      }
+      
+      Section(
+        header: Text(domains.isEmpty ? Strings.PageZoom.defaultZoomLevelSectionTitle : Strings.PageZoom.otherWebsiteZoomLevelSectionTitle)
+          .font(.footnote)
+          .foregroundColor(Color(UIColor.secondaryBraveLabel))
+      ) {
+        ForEach(PageZoomView.steps, id: \.self) { step in
+          
+          let zoomLevel = Int(step * 100.0)
+          let preferredZoomLevel = Int(defaultZoomLevel * 100.0)
+         
+          Button(action: {
+            Preferences.General.defaultPageZoomLevel.value = step
+            defaultZoomLevel = step
+            
+            domains = PageZoomSettingsView.filteredDomains(context)
+          }, label: {
+            HStack(spacing: 0.0) {
+              Text(PageZoomView.percentFormatter.string(from: .init(value: step)) ?? "\(zoomLevel)%")
+                .foregroundColor(Color(UIColor.braveLabel))
+                .multilineTextAlignment(.trailing)
+              
+              Spacer()
+              
+              if preferredZoomLevel == zoomLevel {
+                Image(systemName: "checkmark")
+                  .foregroundColor(Color(UIColor.braveLabel))
+              }
+            }
+          })
+        }
+      }
+      .listRowBackground(Color(UIColor.secondaryBraveGroupedBackground))
+      
+      if !domains.isEmpty {
+        Section(
+          header: Text(Strings.PageZoom.resetAllDescription)
+            .font(.footnote)
+            .foregroundColor(Color(UIColor.secondaryBraveLabel))
+        ) {
+          Button(Strings.PageZoom.resetAll) {
+            PageZoomSettingsView.allDomains(context).forEach({
+              $0.zoom_level = nil
+            })
+            
+            try? context.save()
+            
+            domains = PageZoomSettingsView.filteredDomains(context)
+            
+            Preferences.General.defaultPageZoomLevel.notify()
+          }
```

Comment:
Can we move this into a separate function the same way we have the private func for resetting item zoom level?